### DISCUSSION TITLE
Setup the Daily Solar Yield

### DISCUSSION THREAD

**Discussion by pow3r77** [wlcrs/huawei_solar#306] [Setting up the integration]
Hello to everybody, I have setup your integration and work very well!.
I have see that to have correct Daily Yeald (I have also a battery) i need to modify the sensor like in this guide:
https://github.com/wlcrs/huawei_solar/wiki/Daily-Solar-Yield
I have followed all the steps, at the end i suppose i need to change in my Home Assistant the energy tab, in "Solar Panel" section the new sensor "solar_energy_riemann but I can't find it in the avaiulable sensor.
Whats I do wrong? Anyone can help me? Any help is much apreciate.
Thank you in advance.

**Comment by d3zit** [comment]
Same issue

**Reply by d3zit** [reply]
Tried everything. The corrected efficiency sensor works well, but it's impossible to use the integration. It doesn't work even the "uncorrected" version of the integral sensor. It simply doesn't show up as a sensor in the developer tools

**Reply by Roving-Ronin** [reply]
@pow3r77 For it to be an option to select in the Energy Dashboard you need a sensor that includes "device_class: "energy"" in its configuration, or a utility_meter that uses this energy sensor.
